manager/controlapi/service.go

```diff
@@ -525,6 +526,13 @@ func (s *Server) GetService(ctx context.Context, request *api.GetServiceRequest)
 		return nil, grpc.Errorf(codes.NotFound, "service %s not found", request.ServiceID)
 	}
 
+	if request.InsertDefaults {
+		service.Spec = *defaults.InterpolateService(&service.Spec)
+		// Clear version to prevent the returned service from being the
+		// basis for an update.
+		service.Meta.Version = api.Version{}
```

Comment:
I'm having second thought about that.

It would make `docker service inspect` not include the version which is weird